### ISSUE TITLE
Synergies persisting on screen

### ISSUE THREAD

**Issue by Wheel5** [Wheel5/Synergy#4]
observed with black widows synergy, synergy remained on screen after pull in vhof until a new synergy became available (orbs). unsure if caused by addon or issue in zos function, perhaps an attempted block ended up killing a function call before the ui element could be hidden?

**Comment by Supportic** [comment]
I forgot to implement the check if n exists before it's converted to a string.
synergyName is declared as nilable, therefore you get "empty string" if you convert a nil value.

```lua
if n then
  -- solve localization issue
  n = zo_strformat("<<1>>", n)
end
```

Please rename the dps role into dd instead of d, it was a pain to understand that my debug function is not working. 💃 

Also I suggest to add a debug option in the menu to display the synergy name when a new is available.
Declare a new var called `syn.oldSynergyName = ""`.

```lua
if n then
  -- solve localization issue
  n = zo_strformat("<<1>>", n)

  if syn.oldSynergyName ~= n then
    syn.oldSynergyName = n;
    d(n)
  end
end
```

**Comment by Wheel5** [comment]
added the checks in efb84f20, might add the debug options in the menu in the future. Still not certain this was the issue, as even though `n` wouldn't be nil, it wouldn't match any of the cases I've made, and then zos' [original function](https://github.com/esoui/esoui/blob/360dee5f494a444c2418a4e20fab8237e29f641b/esoui/ingame/synergy/synergy.lua#L58) would be run and they grab their own version of the name in there, so it shouldn't matter anymore.

**Comment by Supportic** [comment]
I couldn't activate the synergy as magicka dps when I was in a group. Outside of the group it was possible to activate it. Maybe it has something to do before entering the combat. The synergy range is huge of this skill and therefore it's still available. Just assumptions.

**Comment by Wheel5** [comment]
You were able to to use the synergy because when you are not in a group, `local dd, h, t = GetGroupMemberRoles('player')` returns `false` for all 3 roles (as far as I remember from my testing at least). I don't consider this a bug since if you aren't in a group, you probably also don't really care about whether or not you can take group oriented synergies anyway ;)

**Comment by Supportic** [comment]
> You were able to to use the synergy because when you are not in a group, local dd, h, t = GetGroupMemberRoles('player') returns false for all 3 roles (as far as I remember from my testing at least). 

Yes, this is true. I tried to reproduce the situation. I wonder which role you had assigned while encountering this bug.

**Comment by Wheel5** [comment]
I was in halls of fabrication as a dps, and there was definitely no synergy actually available. Haven't seen it since, and it only happened once the whole night, so idk. Will be raiding with this patch tonight, and if nothing weird happens I'm just gonna release it.
